T/t8code/build_tarballs.jl

```diff
@@ -0,0 +1,101 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "t8code"
+version = v"1.1.2"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/DLR-AMR/t8code/releases/download/v$(version)/t8code_v$(version).tar.gz", "0bd4bee6694735d14fb4274275fb8c4bdeacdbd29b257220c308be63e98be8f7")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd t8code/
+
+# Set default preprocessor and linker flags
+# Note: This is *crucial* for Windows builds as otherwise the wrong libraries are picked up!
+export CPPFLAGS="-I${includedir}"
+export LDFLAGS="-L${libdir}"
+export CFLAGS="-O3 -std=c99"
```

Comment:
Sounds like the upstream package should set the standard flags. Users should never have to do that themselves.

Reply:
In general I agree. Support for Windows on t8code is still in its infancy and will be improved in the future. For now, we want to focus on shipping a `t8code_jll` for all major platforms.

Reply:
@sloede By sophisticated way, you mean having `MPITrampoline` as an dependency?

Reply:
@jmark Yes, basically trying to use a setup like it is used for p4est:
https://github.com/JuliaPackaging/Yggdrasil/blob/d55ac06f2ab98afc070f1e0e6fd5a5d300578c85/P/P4est/build_tarballs.jl#L66-L89

@giordano We are using an ancient compiler here that is not officially supported by t8code. I'll try switching to a minimum gcc v8 (like we do for p4est), which might allow us to remove these flags.

Reply:
That's good, but if a source code requires a certain language standard, its build system should set it up correctly, whatever is the default standard used by the compilers. Mine was just a general comment, outside of the scope of this PR.

Reply:
@giordano We we able to remove the language standard flags after switchting to GCC 8.